### ISSUE TITLE
Feature request: read a large DataFrame in chunks

### ISSUE THREAD

**Issue by Crypto7816** [man-group/ArcticDB#1709]
**Is your feature request related to a problem? Please describe.**
If we have a really large dataframe that exceeds memory, and we need to process each part of it, `parquet` supports `batch_size`. I'm wondering if `lib.read` has similar functionality.
```python
def read_parquet_in_batches(file_path, batch_size=10000):
    parquet_file = pq.ParquetFile(file_path)
    for batch in parquet_file.iter_batches(batch_size=batch_size):
        yield batch.to_pandas()
```

**Comment by DrNickClarke** [comment]
Hi. You can do this using the `row_range` argument of the `read` function.

We are planning various future improvements that will make this easier to use and faster.

**Comment by Crypto7816** [comment]
> Hi. You can do this using the `row_range` argument of the `read` function.
> 
> We are planning various future improvements that will make this easier to use and faster.

Yes, it can indeed be done using `row_range` or `date_range`，I want a function that's like this：

```python
def fetch_batch_from_arcticdb(
    symbol: str,
    start: str,
    end: str,
    batch_size: int = 1440,
    uri: str = 'lmdb://crypto_database.lmdb',
    library: str = 'binance',
):
    ac = adb.Arctic(uri)
    lib = ac[library]
    
    start_date = pd.Timestamp(start)
    end_date = pd.Timestamp(end)
    
    while start_date < end_date:
        batch_end = min(start_date + timedelta(minutes=batch_size), end_date)
        
        df = lib.read(symbol, date_range=(start_date, batch_end)).data
        
        yield df
        
        start_date = batch_end
```

**Comment by Crypto7816** [comment]
> Hi. You can do this using the `row_range` argument of the `read` function.
> 
> We are planning various future improvements that will make this easier to use and faster.

Hi, any new updates?

**Comment by vasil-pashov** [comment]
> > Hi. You can do this using the `row_range` argument of the `read` function.
> > We are planning various future improvements that will make this easier to use and faster.
> 
> Hi, any new updates?

Hi, the road map is not completely sorted out. @DrNickClarke will get back to you with more info.

**Comment by DrNickClarke** [comment]
Hi. Sorry for the delay coming back on this. Thank you for your suggestion. We definitely have plans to make chunking easier going forward. It has not reached the top of the priority list at this time but I hope you will be pleased to see the announcements we will be making in the near future.

**Comment by Crypto7816** [comment]
> Hi. Sorry for the delay coming back on this. Thank you for your suggestion. We definitely have plans to make chunking easier going forward. It has not reached the top of the priority list at this time but I hope you will be pleased to see the announcements we will be making in the near future.

I find some great features in [dask](https://docs.dask.org/en/stable/generated/dask.dataframe.read_sql_query.html#dask.dataframe.read_sql_query). It would be great if arctic can implement it.

**Comment by DrNickClarke** [comment]
Hi. We managed to find time to add a basic chunking api. Here is the PR with the new api and tests. We would value your feedback.

https://github.com/man-group/ArcticDB/pull/1853
